src/confd/src/ietf-system.c

```diff
@@ -344,28 +346,36 @@ static int change_clock(sr_session_ctx_t *session, uint32_t sub_id, const char *
 	timezone = srx_get_str(session, XPATH_BASE_"/clock/timezone-name");
 	if (!timezone && !tz_utc_offset) {
 		snprintf(tz_name,sizeof(tz_name),"Etc/UTC");
-		timezone = tz_name;
+	        tz_ptr = tz_name;
 	}
 
 	if (tz_utc_offset) {
 		int8_t offset = atol(tz_utc_offset);
 		/* When using Etc/GMT offsets, the +/- is inverted in tzdata. */
 		snprintf(tz_name,sizeof(tz_name), "Etc/GMT%s%.2d", offset>-1?"":"+", -offset);
-		timezone = tz_name;
+		tz_ptr = tz_name;
+	}
+	else if (timezone) {
+		tz_ptr = timezone;
 	}
```

Comment:
Rather than maintaining both heap- and stack versions of the tz name and offset, I think it could be easier to `asprintf()` in the default values in case they are not available in `running`. Less pointer juggling :juggling_person: